tmt/steps/provision/__init__.py

```diff
@@ -200,6 +462,30 @@ def start(self) -> None:
         """
         self.debug(f"Doing nothing to start guest '{self.guest}'.")
 
+    # A bit of Python magic: we want to run `sync()` when first accessing
+    # `self.facts`. But the field might be already initialized because
+    # of how guest data => guest key loading works. Also, we can't afford
+    # to have an extra member, like ``_facts``, to store ``None``until facts
+    # are loaded, because such a field would not be serialized, and we want
```

Comment:
> because such a field would not be serialized

I don't follow this part. What are we trying to serialize here? This Python magic and the long comment is regarding `Guest.facts` property, where the  `Guest` class is derived from just `tmt.utils.Common`, so it looks like it's not even intended for serialization. So what would be wrong about `Guest._facts` property?

Reply:
Rephrased in 1e603a23f05f7aa894c5ab5875acf049e8548ed0, does my jibberish make any sense now? :)

Reply:
@janhavlin PTAL pls